pkg/registry/extensions/daemonset/strategy.go

```diff
@@ -77,6 +78,14 @@ func (daemonSetStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, ol
 	if !reflect.DeepEqual(oldDaemonSet.Spec, newDaemonSet.Spec) {
 		newDaemonSet.Generation = oldDaemonSet.Generation + 1
 	}
+
+	// update is not allowed to set TemplateGeneration
+	newDaemonSet.TemplateGeneration = oldDaemonSet.TemplateGeneration
+
+	// Any changes to the Spec.Template increment the TemplateGeneration number.
+	if !reflect.DeepEqual(oldDaemonSet.Spec.Template, newDaemonSet.Spec.Template) {
```

Comment:
Move it before? I don't understand? I may move inside the previous `if` clause to be more optimal.